tutorpicasso/commands/run_extra_commands.py

```diff
@@ -0,0 +1,157 @@
+import re
+import subprocess
+from itertools import chain
+
+# Was necessary to use this for compatibility with Python 3.8
+from typing import Any, List
+
+import click
+from tutor import config as tutor_config
+
+COMMAND_CHAINING_OPERATORS = ["&&", "&", "||", "|", ";"]
+
+
+@click.command(name="run-extra-commands", help="Run tutor commands")
+def run_extra_commands() -> None:
+    """
+    This command runs tutor commands defined in PICASSO_EXTRA_COMMANDS
+    """
+    context = click.get_current_context().obj
+    tutor_conf = tutor_config.load(context.root)
+
+    picasso_extra_commands: Any = tutor_conf.get("PICASSO_EXTRA_COMMANDS", None)
+
+    if not picasso_extra_commands:
+        return
+
+    error_message = validate_commands(picasso_extra_commands)
+    if error_message:
+        raise click.ClickException(error_message)
+
+    list(map(run_command, picasso_extra_commands))
+
+
+def validate_commands(commands: Any) -> str:
+    """
+    Takes all the extra commands sent through config.yml and verifies that
+    all the commands are correct before executing them
+
+    Args:
+        commands (list[str] | None): The commands sent through PICASSO_EXTRA_COMMANDS in config.yml
+    """
+    splitted_commands = [
+        split_string(command, COMMAND_CHAINING_OPERATORS) for command in commands
+    ]
+    flat_commands_list: chain[str] = chain.from_iterable(splitted_commands)
+
+    invalid_commands = []
+    misspelled_commands = []
+    for command in flat_commands_list:
+        if "tutor" not in command.lower():
+            if find_tutor_misspelled(command):
+                misspelled_commands.append(command)
+            else:
+                invalid_commands.append(command)
+
+        error_message = ""
+
+        if invalid_commands:
+            error_message += (
+                f"Found some issues with the commands:\n\n"
+                f"=> Invalid commands: {', '.join(invalid_commands)}\n"
+            )
+
+        if misspelled_commands:
+            error_message += (
+                f"=> Misspelled commands: {', '.join(misspelled_commands)}\n"
+            )
+
+        if error_message:
+            error_message += (
+                "Take a look at the official Tutor commands: "
+                "https://docs.tutor.edly.io/reference/cli/index.html"
+            )
+            return error_message
+    return ""
+
+
+def run_command(command: str) -> None:
+    """
+    Run an extra command.
+
+    This method runs the extra command provided.
+
+    Args:
+        command (str): Tutor command.
+    """
+    try:
+        with subprocess.Popen(
+            command,
+            shell=True,
+            executable="/bin/bash",
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        ) as process:
+
+            stdout, stderr = process.communicate()
+
+            if process.returncode != 0 or "error" in stderr.lower():
```

Comment:
Is there a case where the process returns 0 but still has an error?

Reply:
I'm not sure. It shouldn't. But it depends on how Tutor handles the errors in its commands.

@bra-i-am do you remember why do we add this?

Ref: https://github.com/eduNEXT/tutor-contrib-edunext-distro/pull/68

Reply:
holi, there were some cases where the extra commands failed, but the execution of `subprocess` returned 0 code (a little bug). For instance, if the `DISTRO_THEMES` config.yml attribute has a typo in the path to the repository and it doesn't apply the text validation, the execution will run as if nothing happened